dolweb/lobby/views.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (c) 2019 Dolphin Emulator Website Contributors
+# SPDX-License-Identifier: MIT
+
+from django.http import HttpResponseBadRequest, HttpResponseNotFound, JsonResponse
+from django.conf import settings
+
+import time
+import random
+import re
+
+secret_chars = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789'
+secret_length = 10
+
+session_valid_keys = ['name', 'region', 'game', 'password', 'method', 'server_id', 'version', 'player_count', 'in_game', 'port']
+
+sessions = {}
+sessions_last_cleanup = 0
+
+def _validate_session(session):
+    # Abort if an unsupported field is provided
+    for key in session:
+        if key not in session_valid_keys:
+            return False
+
+    # Check if all fields are provided
+    for key in session_valid_keys:
+        if key not in session:
+            return False
+
+    if not 0 < len(session['name']) <= settings.SESSION_MAX_NAME_LENGTH:
+        return False
+
+    if not (session['password'] == '1' or session['password'] == '0'):
+        return False
+
+    if not (session['in_game'] == '1' or session['in_game'] == '0'):
+        return False
+
+    if not 0 < int(session['port']) <= 65355:
+        return False
+
+    return True
+
+def _sessions_cleanup():
+    to_delete = []
+    for key in sessions:
+        if time.time() - sessions[key]['timestamp'] > settings.SESSION_TIMEOUT_SECONDS:
+            to_delete.append(key)
+
+    for key in to_delete:
+        del sessions[key]
+
+def _filter_string(sessions, key, value, match = False):
+    filtered_sessions = []
+
+    for session in sessions:
+        if match:
+            if re.match('.*' + value + '.*', session[key]):
+                filtered_sessions.append(session)
+        else:
+            if session[key] == value:
+                filtered_sessions.append(session)
+
+
+    return filtered_sessions
+
+def _bad_method():
+    return JsonResponse({'status':'BAD_METHOD'})
```

Comment:
For consistency with `_bad_version` (and all the others) you should add some spaces here (or remove them over there)